lib/Sema/ConstraintSystem.h

```diff
@@ -296,6 +296,9 @@ class TypeVariableType::Implementation {
 
   /// Retrieve the generic parameter opened by this type variable.
   GenericTypeParamType *getGenericParameter() const;
+  
+  /// Check if there is a generic parameter opened by this type variable.
+  bool hasGenericParameter() const;
```

Comment:
Could you please rename `has` to `is` here because in such cases type variable represents a generic parameter?

Reply:
Humm, true ... sure I can change that :)